test/common/user_dedicated_admin_synced_sso.go

```diff
@@ -0,0 +1,431 @@
+	"strconv"
+	"strings"
+	"time"
+)
+
+const (
+	clientIDAdminCLI                      = "admin-cli"
+	grantTypePassword                     = "password"
+	grantTypeRefreshToken                 = "refresh_token"
+	groupNameDedicatedAdmins              = "dedicated-admins"
+	groupNameDedicatedAdminsRealmManagers = "realm-managers"
+	groupNameRHMIDevelopers               = "rhmi-developers"
+	realmNameMaster                       = "master"
+	testUserName                          = "test-user99"
+)
+
+var testUser = &userv1.User{
+	ObjectMeta: metav1.ObjectMeta{
+		Name: testUserName,
+	},
+}
+
+func TestDedicatedAdminUsersSyncedSSO(t TestingTB, tc *TestingContext) {
+	ctx := context.Background()
+	defer cleanUpTestDedicatedAdminUsersSyncedSSO(ctx, tc.Client)
+
+	// Create Testing IDP
+	if err := createTestingIDP(t, ctx, tc.Client, tc.KubeConfig, tc.SelfSignedCerts); err != nil {
+		t.Fatalf("error creating testing IDP: %v", err)
+	}
+
+	// Create OpenShift user
+	if err := tc.Client.Create(ctx, testUser); err != nil {
+		t.Fatalf("error creating openshift user: %v", err)
+	}
+	t.Logf("openShift user %s created", testUser.Name)
+
+	// Authenticate with Keycloak
+	hostKeycloakUserSSO, err := getHostKeycloak(ctx, tc.Client, RHSSOUserProductNamespace)
+	credentialsKeycloak, err := getCredentialsRHSSOUser(ctx, tc.Client)
+	if err != nil {
+		t.Fatalf("%v", err)
+	}
+	tokenOptions := keycloakTokenOptions{
+		ClientID:  pointer.StringPtr(clientIDAdminCLI),
+		GrantType: pointer.StringPtr(grantTypePassword),
+		RealmName: realmNameMaster,
+		Username:  pointer.StringPtr(credentialsKeycloak[0]),
+		Password:  pointer.StringPtr(credentialsKeycloak[1]),
+	}
+	tokens, err := getKeycloakToken(tc.HttpClient, hostKeycloakUserSSO, tokenOptions)
+	if err != nil {
+		t.Fatalf("%v", err)
+	}
+
+	// Create a testing user in Keycloak (user-sso)
+	keycloakUserToCreate := keycloakUser{
+		EmailVerified: pointer.BoolPtr(true),
+		Enabled:       pointer.BoolPtr(true),
+		FirstName:     pointer.StringPtr("Test User"),
+		LastName:      pointer.StringPtr("99"),
+		UserName:      pointer.StringPtr(testUserName),
+	}
+	if err := createKeycloakUser(tc.HttpClient, hostKeycloakUserSSO, realmNameMaster, tokens.AccessToken, keycloakUserToCreate); err != nil {
+		t.Fatalf("%v", err)
+	}
+	t.Logf("created Keycloak user with username: %s", testUserName)
+
+	// Add the test user to the dedicated-admins group
+	if err := createOrUpdateDedicatedAdminGroupCR(ctx, tc.Client, []string{testUserName}); err != nil {
+		t.Fatalf("failed to add user %s to group %s: %v", testUserName, groupNameDedicatedAdmins, err)
+	}
+	t.Logf("added user %s to group %s", testUserName, groupNameDedicatedAdmins)
+
+	// Wait for RHOAM to reconcile the KeycloakUser CR in user-sso namespace and verify their group membership
+	if err := pollGeneratedKeycloakUserCR(ctx, tc.Client); err != nil {
+		t.Fatalf("%v", err)
+	}
+
+	// Wait for group memberships to propagate in Keycloak
+	time.Sleep(90 * time.Second)
```

Comment:
I had 3/5 runs of this test failing with: 
```
 Expected user groups [dedicated-admins realm-managers rhmi-developers], got [rhmi-developers]
```
It might be because of the `pollGeneratedKeycloakUserCR` and/or this timeout :thinking:

Reply:
Yes, it's definitely a timeout issue in the highlighted snippet. I couldn't figure out a better solution than using time.Sleep().
I can increase the timer by another 30 seconds and this should be sufficient to consistently pass that test.
What's your advice on this?

Reply:
Hm, I used to believe that the good timeout is twice the expected time. I'm not familiar with the Keycloak propagation time. Is there a way to wait for it with a timeout for something like 5 minutes? The first thing that comes into my mind is recording a timestamp here and then when we compare expected vs got we enter the loop that breaks either when things match or then we reach a timeout :thinking: Something like [this](https://github.com/cecobask/integreatly-operator/blob/MGDAPI-3454/test/common/quota.go#L60-L69) for example :thinking: 
Oh, and about that `poll` thing - If I'm not wrong it isn't working at all. Here is an [example](https://github.com/cecobask/integreatly-operator/blob/MGDAPI-3454/pkg/webhooks/operator_webhooks.go#L274-L284) I've managed to find. As far as I know the `Poll` breaks if we return `true` or an error, so to repeat it we should return `false` and no errors, right? :thinking:

Reply:
I've replaced `time.Sleep()` with `wait.PollImmediate()`. Tested on my cluster and it works as expected. The timeout I set is 5 minutes, as suggested. Thank you for the feedback :)

Reply:
Looks nice!